### PR TITLE
enh(centreonvault): adapt vault module to the new specs

### PR DESCRIPTION
**Pull request by omercier**
## Description

The centreonvault.pm module has been adapted to the new vault specs.

Refs: CTOR-1151

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Functionality enhancement or optimization (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## How this pull request can be tested ?

Please describe the **procedure** to verify that the goal of the PR is matched. 
Provide clear instructions so that it can be **correctly tested**.
Mention the automated tests included in this FOR (what they test like mode/option combinations).

## Checklist

- [x] I have followed the **[coding style guidelines](https://github.com/centreon/centreon-plugins/blob/develop/doc/en/developer/plugins_global.md#5-code-style-guidelines)** provided by Centreon
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (develop).
- [x] In case of a new plugin, I have created the new packaging directory accordingly.
- [x] I have implemented automated tests related to my commits.
- [x] I have reviewed all the help messages in all the .pm files I have modified.
  - [x] All sentences begin with a capital letter.
  - [x] All sentences are terminated by a period.
  - [x] I am able to understand all the help messages, if not, exchange with the PO or TW to rewrite them.
- [x] After having created the PR, I will make sure that all the tests provided in this PR have run and passed.

### CODE REVIEW

**Review by kduret — DISMISSED** [review]

File: src/centreon/plugins/passwordmgr/centreonvault.pm [hunk 1]
```diff
@@ -25,28 +25,41 @@ use warnings;
...
+my $VAULT_PATH_REGEX = qr/^secret::hashicorp_vault::([^:]+)::(.+)$/;
 
 sub new {
     my ($class, %options) = @_;
     my $self  = {};
     bless $self, $class;
 
     if (!defined($options{output})) {
-        print "Class PasswordMgr: Need to specify 'output' argument.\n";
+        print "Class PasswordMgr: Needs to be passed an 'output' argument of class output.\n";
```
Comment (sdepassio) [inline src/centreon/plugins/passwordmgr/centreonvault.pm:40]: The error message is not clear

File: src/centreon/plugins/passwordmgr/centreonvault.pm [hunk 2]
```diff
@@ -25,28 +25,41 @@ use warnings;
...
         exit 3;
     }
     if (!defined($options{options})) {
-        $options{output}->add_option_msg(short_msg => "Class PasswordMgr: Need to specify 'options' argument.");
+        $options{output}->add_option_msg(short_msg => "Class PasswordMgr: Needs to be passed an 'options' argument of class options.");
```
Comment (sdepassio) [inline src/centreon/plugins/passwordmgr/centreonvault.pm:44]: The error message is not clear

File: src/centreon/plugins/passwordmgr/centreonvault.pm (hunk 1 above)
Comment (omercier) [reply src/centreon/plugins/passwordmgr/centreonvault.pm:40]: This message will appear only if a developer changes the way password managers are called here https://github.com/centreon/centreon-plugins/blob/056589f360e52434923a86456b6df9651666d34d/src/centreon/plugins/script_custom.pm#L177. 
It shouldn't ever happen in production

**Review by kduret — APPROVED** [review]

File: src/centreon/plugins/passwordmgr/centreonvault.pm (hunk 1 above)
Comment (omercier) [inline src/centreon/plugins/passwordmgr/centreonvault.pm:40]:
```suggestion
        print "Class PasswordMgr needs an 'output' argument that must be of type centreon::plugins::output.\n";
```

**Review by sdepassio — DISMISSED** [review]

File: connectors/vmware/src/centreon/script/centreonvault.pm [hunk 3]
```diff
@@ -364,8 +364,8 @@ The expected file format for Centreon Vault is:
 
     {
       "name": "hashicorp_vault",
-      "url": "vault-server.mydomain.com",
-      "salt": "<base64 encoded(<32 bytes long key used to hash the crypted data)>",
+      "url": "vault-server.my_domain.com",
```
Comment (tuntoja) [inline connectors/vmware/src/centreon/script/centreonvault.pm:367]: I have some doubt regarding the use of underscore in a domain name

Comment (omercier) [inline connectors/vmware/src/centreon/script/centreonvault.pm:367]:
```suggestion
      "url": "vault-server.my-domain.com",
```

File: src/centreon/plugins/passwordmgr/centreonvault.pm [hunk 4]
```diff
@@ -25,28 +25,41 @@ use warnings;
...
     bless $self, $class;
 
     if (!defined($options{output})) {
-        print "Class PasswordMgr: Need to specify 'output' argument.\n";
+        print "Class PasswordMgr needs an 'output' argument that must be of type centreon::plugins::output.\n";
         exit 3;
     }
     if (!defined($options{options})) {
-        $options{output}->add_option_msg(short_msg => "Class PasswordMgr: Need to specify 'options' argument.");
+        print "Class PasswordMgr needs an 'output' argument that must be of type centreon::plugins::options.\n";
```
Comment (omercier) [inline src/centreon/plugins/passwordmgr/centreonvault.pm:44]:
```suggestion
        print "Class PasswordMgr needs an 'options' argument that must be of type centreon::plugins::options.\n";
```

**Review by sdepassio — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

**Pull request merged**